docs/guide/images.md

```diff
@@ -225,3 +225,112 @@ The following filters exist in the Imagine library but are not used in eZ Platfo
 ### Custom filters
 
 Please refer to [LiipImagineBundle documentation on custom filters](http://symfony.com/doc/master/bundles/LiipImagineBundle/filters.html#custom-filters). [Imagine library documentation](http://imagine.readthedocs.org/en/latest/) may also be useful.
+
+## Setting placeholder generator
+
+Placeholder generator enables you to download or use predefined image placeholder for any missing image.
+
+`PlaceholderAliasGenerator::getVariation` method generates placeholder (by delegating it to the implementation of `PlaceholderProvider` interface) if original image cannot be resolved and saves it under the original path.
+
+!!! note 
+
+    Width and height of an original image are required to generate proper placeholder. `PlaceholderAliasGenerator` decorates `AliasGenerator` as `AliasGenerator` is the only place where you can find them.  
+
+This chapter includes two implementations of `PlaceholderProvider` interface:
+
+- GenericProvider
+- RemoteProvider
+
+```
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine;
+
+use eZ\Publish\Core\FieldType\Image\Value as ImageValue;
+
+interface PlaceholderProvider
+{
+    /**
+     * Provides a placeholder image path for a given Image FieldType value.
+     *
+     * @param \eZ\Publish\Core\FieldType\Image\Value $value
+     * @param array $options
+     * @return string Path to placeholder
+     */
+    public function getPlaceholder(ImageValue $value, array $options = []): string;
+}
+```
+
+### GenericProvider
+
+`\eZ\Bundle\EzPublishCoreBundle\Imagine\PlaceholderProvider\GenericProvider` generates placeholder with basic information about original image - [example 1](#semantic-configuration). 
```

Comment:
The link is a bit confusing - can you change the examples to low-level headings and link directly to them?